### DISCUSSION TITLE
Problem with yarnpkg-core Rollup Config

### DISCUSSION THREAD

**Discussion by israelsgalaxy** [yarnpkg/berry#5752] [General]
Hi,

I came across a chicken-egg problem while packaging yarnpkg-core from source for Debian.
The bundle [output](https://github.com/yarnpkg/berry/blob/4ccec8074cbbc167871e443ad5a62161310df1cc/packages/yarnpkg-core/rollup.config.js#L34) file is being [imported](https://github.com/yarnpkg/berry/blob/4ccec8074cbbc167871e443ad5a62161310df1cc/packages/yarnpkg-core/sources/tgzUtils.ts#L8) by a file that makes up the bundle.
I found that this [commit](https://github.com/yarnpkg/berry/commit/c4f04c0485909f8216469ee969837602911aebb6) introduced these changes.

Question: How could I generate the bundle without a pre-existing bundle? (a pre-existing bundle can't be used when packaging for Debian since we try to build everything from source)

Thanks.

**Comment by arcanis** [comment]
You can generate the CLI using something like this:

```
yarn build:zip:worker
yarn build:cli+hook
```

This will generate the `packages/yarnpkg-cli/bundles/yarn.js` file, and the zip worker will also be regenerated from sources.
